src/main/java/org/betonquest/betonquest/commands/QuestCommand.java

```diff
@@ -1695,6 +1702,59 @@ private void handleDebug(final CommandSender sender, final String... args) {
         sendMessage(sender, "unknown_argument");
     }
 
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private void handleDownload(final CommandSender sender, final String... args) {
+        if (args.length < 5) {
+            sendMessage(sender, "arguments");
+            return;
+        }
+        final String githubNamespace = args[1];
+        final String ref = args[2];
+        final String offsetPath = args[3];
+        final String sourcePath = args[4];
+        final String targetPath = args.length < 6 ? sourcePath : args[5];
+        final String errSummary = String.format("Download from %s %s of %s from %s to %s failed:",
+                githubNamespace, ref, offsetPath, sourcePath, targetPath);
+
+        //Check offset paths
+        if (!Set.of("QuestPackages", "QuestTemplates").contains(offsetPath)) {
+            sendMessage(sender, "download_failed", "Invalid offset path.");
+            LOG.warn(errSummary, new IllegalArgumentException(offsetPath));
+            return;
+        }
+
+        //check if repo is allowed
+        final List<String> whitelist = instance.getPluginConfig().getStringList("download.repo_whitelist");
+        if (whitelist.stream().map(String::trim).noneMatch(githubNamespace::equals)) {
+            sendMessage(sender, "download_failed", "Repository not whitelisted.");
+            LOG.warn(errSummary, new IllegalArgumentException(githubNamespace));
+            return;
+        }
+
+        //check if ref is valid
+        if (ref.toLowerCase(Locale.ROOT).startsWith("refs/pull/") && !instance.getPluginConfig().getBoolean("download.pullrequests", false)) {
+            sendMessage(sender, "download_failed", "Downloading of pull requests is disabled to prevent exploits.");
+            LOG.warn(errSummary, new IllegalArgumentException(ref));
+            return;
+        }
+
+        //run download
+        final boolean recursive = args.length >= 7 && Boolean.parseBoolean(args[6]);
+        final boolean override = args.length >= 8 && Boolean.parseBoolean(args[7]);
+        final Downloader down = new Downloader(instance.getDataFolder(), githubNamespace, ref, offsetPath, sourcePath, targetPath, recursive, override);
```

Comment:
i am not a big fan of shorten variables like `downloader` to `down`, because it is not that clearly